src/aggregation/segment_agg_result.rs

```diff
@@ -177,10 +186,20 @@ impl SegmentBucketResultCollector {
             SegmentBucketResultCollector::Range(range) => {
                 range.collect_block(doc, bucket_with_accessor, force_flush);
             }
+            SegmentBucketResultCollector::Histogram(histogram) => {
+                histogram.collect_block(doc, bucket_with_accessor, force_flush)
+            }
         }
     }
 }
 
+#[derive(Clone, Debug, PartialEq)]
+pub(crate) struct SegmentHistogramBucketEntry {
+    pub key: f64,
+    pub doc_count: u64,
+    pub sub_aggregation: Option<SegmentAggregationResultsCollector>,
```

Comment:
I wonder if we should also keep sub_aggregation as an external `Vec<Option<SegmentAggregationResultsCollector>>`.

Reply:
yes, that's something I wanted to test as well, but I would probably go for `Option<Vec<SegmentAggregationResultsCollector>>`

Reply:
Oh right.

Reply:
Yep, it's much faster like that, 2x in some cases